src/components/data_display/routes.ts

```diff
@@ -152,24 +156,80 @@ const routeDataViewFilterList = async (request: any, reply: any) => {
 
   const dataView = dataViews[0]
 
-  // Check column exists on highest priority dataView
-  if (!dataView.tableViewIncludeColumns?.includes(columnName))
-    throw new Error(`Column "${dataViewCode}" not available on data view: "${dataView.title}"`)
+  // TO-DO: Create search filters for types other than string (number, bool, array)
+  const searchFilter =
+    searchText === ''
+      ? {}
+      : {
+          or: searchFields.map((col: string) => ({
+            [col]: {
+              includesInsensitive: searchText,
+            },
+          })),
+        }
+
+  const gqlFilters = { ...getFilters(dataView, userId, orgId), ...searchFilter }
+
+  const filterList = new Set()
 
-  // GraphQL query column including row_restrictions
-  const gqlFilters = getFilters(dataView, userId, orgId)
+  let fetchedCount = 0
+  let offset = 0
+  let moreResultsAvailable = true
 
-  const items = (
-    await queryFilterList(
+  while (filterList.size < config.filterListMaxLength) {
+    const { fetchedRecords, totalCount, error } = await queryFilterList(
```

Comment:
Fetch a few records at a time until we reach the "maxListLength" (currently set to 10). We can't just fetch 10 records though, as it's not a one-to-one mapping, due to:
- we want 10 *distinct* values, which might take more than 10 records to find
- some single records will be split into multiple values, using the "delimiter" parameter